include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -345,7 +373,7 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 									for (auto& x = localCoord[0] = 0u; x < state->extent.width; ++x)
 									{
 										const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(localCoord, scratchByteStrides);
-										decodeType* entryScratchAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
+										decodeType* entryScratchAdress = reinterpret_cast<decodeType*>(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
 
 										normalize(entryScratchAdress);
```

Comment:
your use of lambdas just made this function slow... if you just did it in-place with if statements if coult actually be optimized away

Also, `normalizeAsUsual` should be the same as `normalizeAsUnsigned`

Reply:
okay, corrected in following commit